osu.Game/Screens/BackgroundScreen.cs

```diff
@@ -37,10 +37,17 @@ public override void Push(Screen screen)
             }
 
             // Make sure the in-progress loading is complete before pushing the screen.
-            while (screen.LoadState < LoadState.Ready)
+            while (screen.LoadState != LoadState.Ready)
```

Comment:
So what happens if you push a screen that is already in the `Loaded` state? This will infinite loop.

Reply:
This can't happen with the current usage of `BackgroundScreen`. We can add an exception before this if that makes it feel better.

Reply:
Yeah an exception would be good.

Reply:
See ppy/osu-framework#1655

Reply:
There wasn't a change for this in the framework PR.